pages/index.tsx

```diff
@@ -199,17 +200,43 @@ const Home: NextPage = () => {
           Get started by editing <code className={classes.code}>pages/index.js</code>
         </p>
 
-        <BlogSection posts={FEATURED_DUMMY_DATA} variation="featured" />
-        <BlogSection title="Most Popular" posts={MOST_POPULAR_DUMMY_DATA} variation="grid" />
-        <BlogSection
-          title="Browse All"
-          posts={BROWSE_ALL_DUMMY_DATA}
-          variation="grid"
-          dropdown={true}
-        />
+        <BlogSection posts={posts} variation="featured" />
+        <BlogSection title="Most Popular" posts={posts} variation="grid" />
+        <BlogSection title="Browse All" posts={posts} variation="grid" dropdown={true} />
       </main>
     </div>
   );
 };
 
+export async function getStaticProps() {
+  const response = await fetchEntries();
+  const posts = await transformContentfulPosts(response);
+
+  return {
+    props: {
+      posts,
+    },
+  };
+}
+
+const transformContentfulPosts = (items: object[] | undefined): IBlogPost[] | void => {
+  if (!items) {
+    return;
+  }
+
+  const posts = items.map((item: object): IBlogPost => {
```

Comment:
The option 2 is to use a default value for your items object. That way you can always trust the map operator is going to work.

```ts
// Use the default value syntax for the items.
// This means, if the items is undefined or null, we use the default empty array.
const transformContentfulPosts = (items: object[] = []): IBlogPost[] => {
}
```

Besides that, mapping an empty array also produces an empty array as a result:
```ts
  const x = [ ].map( /* whatever function */ );
  // x contains [ ]
```

Maybe this option is better.